### ISSUE TITLE
Proact-VL: Doubts about loss_active_scale and resizing vocabulary

### ISSUE THREAD

**Issue by kffmta** [microsoft/AnthropomorphicIntelligence#63]
Thank you for the great work and releasing the source. 
I am trying to reproduce the results based on Qwen3-VL but I keep getting Nan loss after few steps, so I am going through the code and have some doubts.
1. The loss_active_scale in train.py(line 131) is commented out, why is it so? Shouldn’t it be uncommented?
2. In the file modeling_proact.py when you add the special 9 tokens(Line 93-96), where do you resize the model's token embedding layer to match the tokenizer's vocabulary size?   
Thank You

**Comment by Kelvin-ywc** [comment]
Thank you for your interest in our work, and sorry for the late reply.

For the first question, thanks for pointing this out. Your observation is correct and helped us identify an issue in the current released code. We originally planned to migrate this argument from `ModelArguments` to `TrainingArguments` when cleaning up the code, but due to our oversight, this parameter was not correctly passed to the model.

A quick workaround is to uncomment `train.py` line 131 so that `loss_active_scale` is properly passed to the model. We are updating the code to fix this issue and avoid similar misunderstandings.

For the second question: for Qwen-series models, the embedding matrix is usually larger than the tokenizer's actual vocabulary size because Qwen reserves some placeholder/padding token slots. For example, in our check, Qwen3-VL has around `151,669` actual tokenizer tokens, while its `config.vocab_size` / embedding rows are `151,936`, leaving about `267` reserved slots. Therefore, we do not explicitly call `resize_token_embeddings` after adding the special tokens. Instead, these special tokens are assigned to the reserved placeholder slots.

Regarding the NaN loss after a few steps, our current analysis suggests that it may be caused by `loss_active_scale` not being correctly passed to the model. In the released code, the default value is `1.0`, while the value used in our paper is `0.2`. Please try the workaround above first. If the NaN loss still occurs, could you please share your training script or command, including the key hyperparameters? That would help us better locate the issue.

**Comment by kffmta** [comment]
Thank you for your response.
I doubt the issue for Nan is the `loss_active_scale` being 1, but is more likely due to GPU memory being at the extreme when running the 8B one, because both 2B and 4B run just fine. And when I use deepspeed zero3 the 8B one also seems to be running fine. My runs are only done on the YuGiOh dataset. I will soon try to run on the entire data and see if with Zero3 it works.

**Comment by kffmta** [comment]
Additionally I had one more questions, how large is the full training data used?

**Comment by Kelvin-ywc** [comment]
The full training set contains about 128K training samples. For most videos except the Minecraft videos, each sample is around 36 seconds long. For the Minecraft videos, we process the video samples into both 36-second and 60-second clips.

The size of the training set, [game_commentary_sft](https://huggingface.co/datasets/oaaoaa/game_commentary_sft), is about 152 GB. In addition, you need to separately apply for the [Ego4D-GoalStep](https://github.com/facebookresearch/ego4d-goalstep), which is around 500 GB, and download the first 32,000 chunks of [Live-WhisperX-526K](https://huggingface.co/datasets/chenjoya/Live-WhisperX-526K). Specifically, we use the chunks from videos_range0-3_chunk0-2000.tar to videos_range0-3_chunk30000-32000.tar, which are about 400 GB in total.

So the full data required for training is roughly around 1 TB.

**Comment by kffmta** [comment]
I ran the full the training using Qwen3-VL, the results are slightly lower that reported in paper. I got F1 score: 58.7, and TimeDiff: 2.29.
I was wondering how did you get the results from offline models like GPT-4o and other baselines? Is it possible to share the scripts for those?
Thank You

**Comment by Kelvin-ywc** [comment]
Thank you for reproducing the results and for sharing the numbers.

In our experiments, response quality is quite sensitive to the response threshold, and different inference configurations can also affect the final evaluation results. For the main results in the paper, we use a response threshold of 0.3.

Could you first try evaluating our released checkpoint, `oaaoaa/proactvl_base_qwen3vl`? This would help us better determine whether the discrepancy mainly comes from the training stage or the evaluation/inference setup.

For the offline-model baselines, we have prepared scripts for generating two types of labels:

1. **Caption labels**: given a video segment, the model generates a caption for the entire clip. These outputs are used to compute the CC score, following the evaluation protocol in LiveCC.

2. **Interleaved response labels**: each output contains both the response timestamp and the corresponding response content. These outputs are used for the evaluation metrics other than CC.

**Caption generation:**

```bash
python -m evaluation.gaming.distributed_generate_gaming_caption \
    --model_name_or_path openai/gpt-4o-2024-11-20 \
    --game_list baldurs_gate_3 black_myth_wukong csgo cyberpunk_2077 ego4d elden_ring lol minecraft starcraft2 streetfighter6 tears_of_the_kingdom yu_gi_oh \
    --test_name all_in_one \
    --num_workers 1 \
    --output_dir ./results/baseline
```

**Interleaved response generation:**

```bash
python -m evaluation.gaming.distributed_generate_gaming_interleave \
    --model_name_or_path openai/gpt-4o-2024-11-20 \
    --game_list baldurs_gate_3 black_myth_wukong csgo cyberpunk_2077 ego4d elden_ring lol minecraft starcraft2 streetfighter6 tears_of_the_kingdom yu_gi_oh \
    --test_name all_in_one \
    --num_workers 1 \
    --output_dir ./results/baseline
```

(The labels used for the paper were generated with `gpt-4o-2024-12-01-preview`. This exact model version does not appear to be available through OpenRouter, so the provided scripts use `openai/gpt-4o-2024-11-20` instead.)

We have submitted a pull request containing these updates, and it will be merged into the `main` branch shortly.

Sorry for the delayed reply, and thank you for your careful evaluation and for raising this question.

**Comment by kffmta** [comment]
Thank you for the detailed response. I will check these scripts in some time.
Regarding the results reproduced, I compared my results with this file from your hf repo (https://huggingface.co/datasets/oaaoaa/proactvl_results/blob/main/results/proactvl/qwen3vl_30_16384.jsonl). Please let me know if this is the right file to compare to, for the Qwen3-VL baseline. The results are almost same. So, I have reproduced the results. But I still have no clue why deepspeed2-zero was giving me Nans on the very first update step.

**Comment by Kelvin-ywc** [comment]
Thank you for the clarification.

For computing F1 and TimeDiff, the reference file (--reference) should be:

https://huggingface.co/datasets/oaaoaa/proactvl_results/blob/main/results/anns/all_in_one_val_proactive.jsonl

This file contains the ASR-based human commentary annotations and serves as the ground-truth reference.

Regarding the DeepSpeed ZeRO-2 NaN issue, I am not fully sure about the exact cause. It may be related to a reused or conflicting DeepSpeed/Accelerate config, inconsistent fp16/bf16 settings, loss scaling, optimizer instability, or gradient explosion at the first update step. To stabilize training, you may try using bf16 consistently, disabling fp16, setting `gradient_clipping` or `max_grad_norm` to `1.0`, reducing the learning rate, and adding warmup steps.

**Comment by kffmta** [comment]
Since I am using Qwen3-VL as the decoder, the file I pointed was the one I used to compare my results against I did not use that as reference. For reference file I used the one that you have suggested.
